src/models/tiles/geometry/geometry-utils.ts

```diff
@@ -1,16 +1,57 @@
 import { values } from "lodash";
 import { Instance } from "mobx-state-tree";
 import { getAssociatedPolygon } from "./jxg-polygon";
-import { isPoint, isPolygon } from "./jxg-types";
+import { isGeometryElement, isPoint, isPolygon } from "./jxg-types";
 import { JXGObjectType } from "./jxg-changes";
 import { logTileChangeEvent } from "../log/log-tile-change-event";
 import { LogEventName } from "../../../lib/logger-types";
 import { GeometryBaseContentModel } from "./geometry-model";
 import { getTileIdFromContent } from "../tile-model";
+import { isFiniteNumber } from "../../../utilities/math-utils";
 import { clueDataColorInfo } from "../../../utilities/color-utils";
 
 export function copyCoords(coords: JXG.Coords) {
-  return new JXG.Coords(JXG.COORDS_BY_USER, coords.usrCoords.slice(1), coords.board);
+  const usrCoords = coords.usrCoords;
+  if (usrCoords.length >=3 ) {
+    const shortCoords: [number,number] = [usrCoords[1],usrCoords[2]];
+    return new JXG.Coords(JXG.COORDS_BY_USER, shortCoords, coords.board);
+  } else {
+    return new JXG.Coords(JXG.COORDS_BY_USER, [0, 0], coords.board);
+  }
+}
+
+// Define some helper functions to work around the typing of board.objectsList as unknown[].
+
+export function getBoardObjectIds(board: JXG.Board): string[] {
+  return Object.keys(board.objects);
+}
+
+export function getBoardObject(board: JXG.Board|undefined, id: string): JXG.GeometryElement|undefined {
+  const obj = board && board.objects[id];
+  return isGeometryElement(obj) ? obj : undefined;
+}
+
+export function forEachBoardObject(board: JXG.Board, callback: (elt: JXG.GeometryElement, index: number) => void) {
+  board.objectsList.forEach((obj, index) => {
+    if (isGeometryElement(obj)) { callback(obj, index); }
+  });
+}
+
+export function findBoardObject(board: JXG.Board, callback: (elt: JXG.GeometryElement) => any):
+    JXG.GeometryElement | undefined {
+  const found = board.objectsList.find(obj => { return isGeometryElement(obj) && callback(obj); } );
+  return isGeometryElement(found) ? found : undefined;
+}
+
+export function filterBoardObjects(board: JXG.Board,
+    callback: (elt: JXG.GeometryElement) => any): JXG.GeometryElement[] {
+  return board.objectsList.filter((obj) => {
+    if (isGeometryElement(obj)) {
+      return callback(obj);
+    } else {
+      return false;
```

Comment:
Isn't this the same as:
```suggestion
  return board.objectsList.filter((obj) => isGeometryElement(obj) && callback(obj));
```